### PR TITLE
Feature handleScroll to detect where user is in documentation

### PR DESCRIPTION
**Pull request by Darkweak**

### CODE REVIEW

File: src/components/layout/DocNav.js [hunk 1]
```diff
@@ -31,6 +31,24 @@ class DocNav extends Component {
     }
   }
 
+  handleScroll() {
+    if (document.querySelectorAll('.Collapsible.submenu__item.open')[0] !== undefined) {
```
Comment (dunglas) [inline src/components/layout/DocNav.js]: You should invert the test and return early to make the code more readable/less complex (guard clause)

File: src/components/layout/DocNav.js [hunk 2]
```diff
@@ -31,6 +31,24 @@ class DocNav extends Component {
...
+      const currentItemOpen = document.querySelectorAll('.Collapsible.submenu__item.open')[0];
+      const childsCurrentItem = currentItemOpen.querySelectorAll('a');
+      const childsInnerPageItem = [];
+      childsCurrentItem.forEach((child, index) => {
```
Comment (dunglas) [inline src/components/layout/DocNav.js]: Use `map` instead.

File: src/components/layout/DocNav.js [hunk 3]
```diff
@@ -31,6 +31,24 @@ class DocNav extends Component {
...
+        childsInnerPageItem[index] = document.getElementById(child.getAttribute('href').split('#')[1]);
+        child.parentElement.classList.remove('current');
+      });
+      for (let i = childsCurrentItem.length - 1; 0 < i; i--) {
+        if (childsInnerPageItem[i].offsetTop < window.scrollY) {
+          childsCurrentItem[i].parentElement.classList.add('current');
+          break;
```
Comment (dunglas) [inline src/components/layout/DocNav.js]: `return`?

File: src/components/layout/DocNav.js [hunk 4]
```diff
@@ -31,6 +33,24 @@ class DocNav extends Component {
...
+  handleScroll() {
+    if (document.querySelectorAll('.Collapsible.submenu__item.open')[0] !== undefined) {
+      const currentItemOpen = document.querySelectorAll('.Collapsible.submenu__item.open')[0];
+      const childsCurrentItem = currentItemOpen.querySelectorAll('a');
+      const childsInnerPageItem = [];
+      childsCurrentItem.forEach((child, index) => {
+        childsInnerPageItem[index] = document.getElementById(child.getAttribute('href').split('#')[1]);
+        child.parentElement.classList.remove('current');
+      });
+      for (let i = childsCurrentItem.length - 1; 0 < i; i -= 1) {
```
Comment (teohhanhui) [inline src/components/layout/DocNav.js]: You could use a `forEach` here.

**Comment by dunglas** [comment]
Merged in #42

**Pull request closed without merging**